src/lua/fio.lua

```diff
@@ -99,13 +99,21 @@ end
 -- pread(size, offset) -> str
 -- pread(buf, size, offset) -> len
 fio_methods.pread = function(self, buf, size, offset)
-    local tmpbuf
+    local no_buf = false
     if not ffi.istype(const_char_ptr_t, buf) then
         offset = size
         size = buf
+        no_buf = true
+    end
+    if (type(size) ~= 'number' or (type(offset) ~= 'number' and offset ~= nil)) then
```

Comment:
Nit: Outermost parentheses are not necessary.

Nit: Personally I find this expression a bit too complex -- and/or, several comparisons, negation of what is expected. What if we would have one more parameter?

What do you think about a step-by-step validation?

```lua
local usage = '<...>'
if type(size) ~= 'number' then
    error(usage)
end
if offset ~= nil and type(offset) ~= 'number' then
    error(usage)
end
```

Reply:
I don't think that dividing the `if` is worth it. Maybe, if we explicitly compare with every format, it would be better:
``` lua
local usage = '...'
if type(size) == 'number' and type(offset) == 'number' then
    usage = nil
end
if type(size) == 'number' and offset == nil then
    usage = nil
end
if (usage) then
    error(usage)
end
```

But I feel that it is too much in this case.

Reply:
Okay.